### DISCUSSION TITLE
Could `SessionStore` not require `Debug`?

### DISCUSSION THREAD

**Discussion by tibbe** [maxcountryman/tower-sessions#225] [Q&A]
I tried to implement `SessionStore` for a

```rust
type Pool = diesel_async::pooled_connection::deadpool::Pool<
    diesel_async::sync_connection_wrapper::SyncConnectionWrapper<
        diesel::sqlite::SqliteConnection>>
```

and that doesn't work as neither `SyncConnectionWrapper` nor `SqliteConnection` implement `Debug`.

Maybe one could add some `Debug` implementations in `diesel` and `async-diesel` but the author of those packages might argue that there's nothing useful to print for things like connections.

Is the `Debug` requirement really needed on `SessionStore`?

**Comment by tibbe** [comment]
I did file a [PR](https://github.com/weiznich/diesel_async/pull/199) to add `Debug` for `SyncConnectionWrapper`. `SqliteConnection` looks a bit trickier.

**Reply by marcaddeo** [reply]
@tibbe I made a `SessionStore` for Diesel in my project and I got around this issue by doing:

```rust
#[derive(derive_more::Debug, Clone)]
pub struct DieselSqliteSessionStore {
    #[debug(skip)]
    database: Pool<SyncConnectionWrapper<SqliteConnection>>,
}
```

You can see the full session store [here](https://github.com/marcaddeo/lowboy/blob/main/src/diesel_sqlite_session_store.rs). The project is in it's early stages, so nothing is well-tested and I'm likely doing things I shouldn't currently 😀 But this might help you with your session store.

**Reply by tibbe** [reply]
> You can see the full session store [here](https://github.com/marcaddeo/lowboy/blob/main/src/diesel_sqlite_session_store.rs). The project is in it's early stages, so nothing is well-tested and I'm likely doing things I shouldn't currently 😀 But this might help you with your session store.

I also wrote one as part of one of my (yet unreleased) projects. I've dumped the code at https://github.com/tibbe/examples/tree/main/tower-sessions-async-diesel so we can compare notes. :)

**Comment by maxcountryman** [comment]
I don't recall if there's a specific reason for `Debug` now (but removing it and running the test suite would at least answer the immediate question of if it's easily possible).
